subprojects/test-runner/test-inhouse-runner/src/main/kotlin/com/avito/android/rule/ActivityScenarioRule.kt

```diff
@@ -9,15 +9,20 @@ import com.avito.android.rule.base.BaseActivityScenarioRule
  * actions that send out intents which need to be verified or stubbed. This is similar to
  * MockitoAnnotations.initMocks.
  */
-class ActivityScenarioRule<T : Activity>(activityClass: Class<T>) : BaseActivityScenarioRule<T>(
+class ActivityScenarioRule<T : Activity>(
+    activityClass: Class<T>,
+    private val stubIntents: Boolean
+) : BaseActivityScenarioRule<T>(
     activityClass = activityClass,
     initialTouchMode = true,
     launchActivity = false,
 ) {
 
     override fun afterActivityLaunched() {
         Intents.init()
-        com.avito.android.test.Intents.stubEverything()
+        if (stubIntents) {
+            com.avito.android.test.Intents.stubEverything()
+        }
```

Comment:
Should we consider this flag in `afterActivityFinished` for  `Intents.release()` ?

Reply:
`Intents.release()` must be called after `Intents.init()`. We can make intents initialization optional at all, but it will break existing tests.